isvcs/container.go

```diff
@@ -625,6 +639,7 @@ func (svc *IService) run() {
 						haltStats <- struct{}{}
 						if len(svc.HealthChecks) > 0 {
 							haltHealthChecks <- struct{}{}
+							haltCustomStats <- struct{}{}
```

Comment:
I think this should be in a  different if block - something that matches the if block that starts the customStats:
```
if svc.customStats != nil {
    haltCustomStats <- struct{}{}
}
```